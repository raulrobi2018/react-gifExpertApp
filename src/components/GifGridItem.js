import React from "react";
import "animate.css";
import PropTypes from "prop-types";

export const GifGridItem = ({title, url}) => {
    return (
        <div className="card animate__animated animate__bounce">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    );
};

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};

export default GifGridItem;
