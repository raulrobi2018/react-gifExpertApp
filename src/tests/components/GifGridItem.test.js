import {shallow} from "enzyme";
import GifGridItem from "../../components/GifGridItem";
import React from "react";

//Para que me de la ayuda de funciones hago esta importaciÃ³n
import "@testing-library/jest-dom";

describe("Testing GifGridItem", () => {
    const title = "Title";
    const url = "https://localhost/algo.jpg";

    test("Should return GifGridItem correctly", () => {
        const wrapper = shallow(<GifGridItem title={title} url={url} />);
        expect(wrapper).toMatchSnapshot();
    });
});
