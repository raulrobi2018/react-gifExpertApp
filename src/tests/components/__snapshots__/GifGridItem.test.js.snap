// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem Should return GifGridItem correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Title"
    src="https://localhost/algo.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
